subprojects/file-watching/src/main/java/org/gradle/internal/watch/registry/FileWatcherRegistry.java

```diff
@@ -21,7 +21,7 @@
 import java.nio.file.Path;
 import java.util.Optional;
 
-public interface FileWatcherRegistry extends Closeable {
+public interface FileWatcherRegistry extends FileWatcherUpdater, Closeable {
```

Comment:
Why would we do this?

Reply:
I want the `WatchingVirtualFileSystem` to only talk to the `FileWatcherRegistry` and not expose the watcher update strategy any more. So I could either add the methods of `FileWatcherUpdater` to the `FileWatcherRegistry` and keep the `FileWatcherUpdater` interface separate. Or I let `FileWatchterRegistry` implement `FileWatcherUpdater` and re-use the interface.

What would you expect?

Reply:
I think it's fine to have similar methods on `FileWatcherRegistry`, but it shouldn't implement the interface itself.

Reply:
I removed the interface from `FileWatcherRegistry`. PTAL!